### PR TITLE
Using the right compare function for FieldPaths

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
We need to compare field paths segment per segment and not use the encoded paths.

### CODE REVIEW

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/OrderTest.java [hunk 1]
```diff
@@ -86,10 +86,10 @@ public void verifyOrder() {
     groups[35] = new Value[] {referenceValue("projects/p1/databases/d1/documents/c1/doc2/c2/doc1")};
     groups[36] = new Value[] {referenceValue("projects/p1/databases/d1/documents/c1/doc2/c2/doc2")};
     groups[37] = new Value[] {referenceValue("projects/p1/databases/d1/documents/c10/doc1")};
-    groups[38] =
-        new Value[] {referenceValue("projects/p1/databases/dkkkkklkjnjkkk1/documents/c2/doc1")};
+    groups[38] = new Value[] {referenceValue("projects/p1/databases/d1/documents/c2/doc1")};
     groups[39] = new Value[] {referenceValue("projects/p2/databases/d2/documents/c1/doc1")};
     groups[40] = new Value[] {referenceValue("projects/p2/databases/d2/documents/c1-/doc1")};
+    groups[40] = new Value[] {referenceValue("projects/p2/databases/d3/documents/c1-/doc1")};
```
Comment (pongad) [inline google-cloud-firestore/src/test/java/com/google/cloud/firestore/OrderTest.java]: Why is 40 assigned twice?

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldPath.java [hunk 2]
```diff
@@ -155,6 +155,6 @@ public String toString() {
 
   @Override
   public int compareTo(@Nonnull FieldPath other) {
-    return getEncodedPath().compareTo(other.getEncodedPath());
+    return super.compareTo(other);
```
Comment (pongad) [inline google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldPath.java]: We can technically just not override this altogether right? If you want the delegation to superclass to be explicit, I'm OK with it too.

File: google-cloud-firestore/src/test/java/com/google/cloud/firestore/OrderTest.java (hunk 1 above)
Comment (schmidt-sebastian) [reply google-cloud-firestore/src/test/java/com/google/cloud/firestore/OrderTest.java]: Thanks for noticing. Fixed.

File: google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldPath.java (hunk 2 above)
Comment (schmidt-sebastian) [reply google-cloud-firestore/src/main/java/com/google/cloud/firestore/FieldPath.java]: I was debating between the two options, but I am fine with removing it.

**Pull request merged**